titiler/endpoints/factory.py

```diff
@@ -677,6 +680,225 @@ def part(
+        ):
+            """Return dataset's basic info or the list of available assets."""
+            with rasterio.Env(**self.gdal_config):
+                with self.reader(src_path.url, **self.reader_options) as src_dst:
+                    if not asset_params.assets:
+                        return src_dst.assets
+                    return src_dst.info(**asset_params.kwargs, **kwargs)
+
+        @self.router.get(
+            "/info.geojson",
+            response_model=Feature,
+            response_model_exclude_none=True,
+            response_class=GeoJSONResponse,
+            responses={
+                200: {
+                    "content": {"application/geo+json": {}},
+                    "description": "Return dataset's basic info as a GeoJSON feature.",
+                }
+            },
+        )
+        def info_geojson(
+            src_path=Depends(self.path_dependency),
+            asset_params=Depends(AssetsBidxParams),
+            kwargs: Dict = Depends(self.additional_dependency),
+        ):
+            """Return dataset's basic info as a GeoJSON feature."""
+            with rasterio.Env(**self.gdal_config):
+                with self.reader(src_path.url, **self.reader_options) as src_dst:
+                    info = {"dataset": src_path.url}
+                    if not asset_params.assets:
+                        info["available_assets"] = src_dst.assets
+                    else:
+                        info["assets"] = {
+                            asset: meta.dict(exclude_none=True)
+                            for asset, meta in src_dst.info(
+                                **asset_params.kwargs, **kwargs
+                            ).items()
+                        }
+                    geojson = utils.bbox_to_feature(src_dst.bounds, properties=info)
+
+            return geojson
+
+    # Overwrite the `/metadata` endpoint because the MultiBaseReader output model is different (Dict[str, cogMetadata])
+    # and MultiBaseReader.metadata() method also has `assets` as a requirement arguments.
+    def metadata(self):
+        """Register /metadata endpoint."""
+
+        @self.router.get(
+            "/metadata",
+            response_model=Dict[str, Metadata],
+            response_model_exclude={"minzoom", "maxzoom", "center"},
+            response_model_exclude_none=True,
+            responses={200: {"description": "Return dataset's metadata."}},
+        )
+        def metadata(
+            src_path=Depends(self.path_dependency),
+            asset_params=Depends(AssetsBidxParams),
+            metadata_params=Depends(self.metadata_dependency),
+            kwargs: Dict = Depends(self.additional_dependency),
+        ):
+            """Return metadata."""
+            with rasterio.Env(**self.gdal_config):
+                with self.reader(src_path.url, **self.reader_options) as src_dst:
+                    return src_dst.metadata(
```

Comment:
Here is `assets` is not set, `src_dst.metadata` will raise an error https://github.com/cogeotiff/rio-tiler/blob/master/rio_tiler/io/base.py#L441

We could change the behavior by updating https://github.com/developmentseed/titiler/blob/67696d9cd208ab45afb3150bf3bc6c40379b17bd/titiler/dependencies.py#L131

to `...` and make `assets` a required argument 🤷‍♂️ 

linked to https://github.com/developmentseed/titiler/pull/230/files#r575544597

Reply:
💡 maybe we could add a small endpoint which sole goal is to return the list of bands/assets, then we can make assets / bands required in the dependencies